Addition of a 'Contribute' section with our Style Guide (#15350)
docker/docker.github.io: contribute/style/grammar.md

---
title: Grammar and style
description: Grammar and style guidelines for technical documentation
keywords: grammar, style, contribute
toc_max: 2
--- 

## Acronyms and initialisms

An acronym is an abbreviation you would speak as a word, for example, ROM (for read only memory). Other examples include radar and scuba, which started out as acronyms but are now considered words in their own right.

An initialism is a type of acronym that comprises a group of initial letters used as an abbreviation for a name or expression. If you were using the acronym in a spoken conversation, you would enunciate each letter: H-T-M-L for Hypertext Markup Language.

### Best practice:

- Spell out lesser-known acronyms or initialisms on first use, then follow with the acronym or initialism in parentheses. After this, throughout the rest of your page or document, use the acronym or initialism alone.

> > ‘You can use Single-Sign-on (SSO) to sign in to Notion. You may need to ask your administrator to enable SSO.’

Reviewer: Question: when we introduce an acronym in parentheses, do we capitalize the spelled-out word, as is done here? Even though we wouldn't normally? (single sign-on)

Author: This is actually a mistake in my draft. It should be single-sign on. Good spot! Will amend.

- Where the acronym or initialism is more commonly used than the full phrase, for example, URL, HTML, you do not need to follow this spell-it-out rule.
- Use all caps for acronyms of file types (a JPEG file).
- Don't use apostrophes for plural acronyms. ✅URLs ❌URL’S
- Avoid using an acronym for the first time in a title or heading. If the first use of the acronym is in a title or heading, introduce the acronym (in parentheses, following the spelled-out term) in the first body text that follows.

## Bolds and italics

Unless you're referring to Ul text or user-defined text, you should not add emphasis to text. Clear, front-loaded wording makes the subject of a sentence clear.

### Best practice:

- Don't use bold to refer to a feature name.
- Use italics sparingly, as this type of formatting can be difficult to read in digital experiences.

> Notable exceptions are titles of books, newspapers or magazines.

Reviewer: More relevant examples?

```suggestion
Notable exceptions are titles of articles, blog posts, or specification documents.
```

## Capitalization

Use sentence case for just about everything. Sentence case means capitalizing only the first word, as you would in a standard sentence.

The following content elements should use sentence case:

- Titles of webinars and events
- Headings and subheadings in all content types
- Calls to action
- Headers in boxed text
- Column and row headers in tables
- Links
- Sentences (of course)

> - Anything in the Ul including navigation labels, buttons, headings

Reviewer: ```suggestion
- Anything in the Ul including navigation labels, buttons, headings. Provided that the corresponding UI label uses sentence case; follow the capitalization style of the UI element.
```

- Never contract a noun with is, does, has, or was as this can make it look like the noun is possessive. Your container is ready. Your container’s ready.

## Dates

You should use the U.S. format of month day, year format for dates: June 26, 2021.

When possible, use the month's full name (October 1, 2022). If there are space constraints, use 3-letter abbreviations followed by a period (Oct. 1. 2022).

## Decimals and fractions

In all UI content and technical documentation, use decimals instead of fractions.

### Best practice:

- Always carry decimals to at least the hundredth place (33.76).

- In tables, align decimals on the decimal point.

- Add a zero before the decimal point for decimal fractions less than one (0.5 cm).

## Lists

Lists are a great way to break down complex ideas and make them easier to read and scan.

### Best practice:

- Bulleted lists should contain relatively few words or short phrases. For most content types, limit the number of items in a list to five.
- Don’t add commas (,) or semicolons (;) to the ends of list items.

>     - Some content types may use bulleted lists that contain 10 items, but it's preferable to break longer lists into several lists, each with its own subheading or introduction.

Reviewer: Remove indentation

```suggestion
- Some content types may use bulleted lists that contain 10 items, but it's preferable to break longer lists into several lists, each with its own subheading or introduction.
```

- Avoid using commas in numbers because they can represent decimals in different cultures. For numbers that are five digits or more, use a space to separate.
    
    
    | Correct | Incorrect |
    | --- | --- |
    | 1000 | 1,000 |
    | 14 586 | 14,586 |
    | 1 390 680 | 1,390,680 |

### Versions

When writing version numbers for release notes, follow the example below:

Version 4.8.2

## Punctuation

### Colons and semicolons

- Use colons to: introduce a list inline in a sentence; introduce a bulleted or numbered list; and provide an explanation.
- Do not use semicolons. Use two sentences instead.

### Commas

- Use the serial or Oxford comma - a comma before the coordinating conjunction (and, or) in a list of three or more things.
- If a series contains more than three items or the items are long, consider a bulleted list to improve readability.

### Dashes and hyphens

- Use the em dash (-) sparingly when you want the reader to pause, to create parenthetical statements, or to emphasize specific words or phrases. Always put a space on either side of the em dash.

> - Use an en dash (-) to indicate spans of numbers, dates, or time.

Reviewer: Finally I know which one to use where 👍🏻 :-D
